### PR TITLE
Fixed #25825 --- Implemented __ne__ for template Origin

### PR DESCRIPTION
**Pull request by jaap3**
https://code.djangoproject.com/ticket/25825

### CODE REVIEW

File: django/template/base.py [hunk 1]
```diff
@@ -151,6 +151,9 @@ def __eq__(self, other):
             self.loader == other.loader
         )
 
+    def __ne__(self, other):
```
Comment (bmispelon) [inline django/template/base.py]: This method is only necessary on Python 2, no?

Comment (jaap3) [inline django/template/base.py]: Correct

Comment (bmispelon) [inline django/template/base.py]: Then we could slap an `if six.PY2` around it, what do you think? That way we'll know to remove it when we drop Python 2 support.

Comment (jaap3) [inline django/template/base.py]: I personally don't think that's necessary. A grep for __ne__ and similar Python 2 methods on the entire Django codebase will find this at that time in the future. But if you want to add it I'm fine with it :)

Comment (bmispelon) [inline django/template/base.py]: You're right. We don't have `if six.PY2` for `__ne__` methods anywhere else so let's keep it consistent.

Comment (charettes) [inline django/template/base.py]: FWIW I made sure we don't forget to remove this method [when we drop support for Python 2](https://code.djangoproject.com/ticket/23919).

File: tests/template_tests/test_origin.py [hunk 2]
```diff
@@ -0,0 +1,24 @@
+from unittest import TestCase
+
+from django.template import Engine
+
+from .utils import TEMPLATE_DIR
+
+
+class OriginTestCase(TestCase):
```
Comment (charettes) [inline tests/template_tests/test_origin.py]: Use `SimpleTestCase` to prevent unnecessary transaction wrapping.

Comment (jaap3) [inline tests/template_tests/test_origin.py]: There's no transaction wrapping for unit test.TestCase is there?

Comment (charettes) [inline tests/template_tests/test_origin.py]: Oops sorry, I missed the import somehow and thought you were using `django.test.TestCase`

**Comment by timgraham** [comment]
merged in c6ea4ed5d23671deee6e736fe1daba2c650a4fbb, thanks!

**Pull request closed without merging**